symbol_exporter/db_access_model.py

```diff
@@ -0,0 +1,126 @@
+import hashlib
+import hmac
+import json
+import os
+from datetime import datetime
+
+import requests
+
+from symbol_exporter.ast_symbol_extractor import version
+
+
+class WebDB:
+    host = "https://cf-ast-symbol-table.web.cern.ch"
+
+    def __init__(self):
+        self.secret_token = os.environ["STORAGE_SECRET_TOKEN"].encode("utf-8")
+
+    def _setup_headers(self, dumped_data, url):
+        headers = {
+            "X-Signature-Timestamp": datetime.utcnow().isoformat(),
+            "X-Body-Signature": hmac.new(
+                self.secret_token, dumped_data.encode(), hashlib.sha256
+            ).hexdigest(),
+        }
+        headers["X-Headers-Signature"] = hmac.new(
+            self.secret_token,
+            b"".join(
+                [
+                    url.encode(),
+                    headers["X-Signature-Timestamp"].encode(),
+                    headers["X-Body-Signature"].encode(),
+                ]
+            ),
+            hashlib.sha256,
+        ).hexdigest()
+        return headers
+
+    def push_symbol_table(self, top_level_name, symbol_table):
+        url = f"/api/v{version}/symbol_table/{top_level_name}"
+        dumped_data = json.dumps(symbol_table, default=make_json_friendly, sort_keys=True)
+        r = requests.put(
+            f"{self.host}{url}",
+            data=dumped_data,
+            headers=self._setup_headers(dumped_data, url=url),
+        )
+        r.raise_for_status()
+
+    def get_current_symbol_table_artifacts(self):
+        all_indexted_pkgs = set()
+        symbol_table_url = f"/api/v{version}/symbol_table"
+        extracted_symbols = requests.get(f"{self.host}{symbol_table_url}").json()
```

Comment:
`host` is not an instance variable, no need to make this an instance method, would make it easier to parallelize/test if extracted as separate function.

Reply:
Hmm, I was hoping to not have to pass that in to every function, would it be reasonable to move host to the init and let it be passed in but defaulted to the current value?

Reply:
It's a class variable so we can access it from the class, no need for the instance - unless we are re-assigning it, in that case it should probably be an instance variable.